cli/unipipe/commands/terraform.ts

```diff
@@ -0,0 +1,196 @@
+      const repository = new Repository(repo ? repo : ".");
+      const out = await run(repository, options);
+      console.log(out);
+    });
+}
+
+export async function run(
+  repo: Repository,
+  opts: TerraformOpts,
+): Promise<string[][]> {
+  console.log("starting terraform command");
+
+  const instances = await repo.mapInstances(async (instance) => {
+    return await instance;
+  });
+
+  return await Promise.all(
+    instances.map((instance) => mapBindings(instance, repo)),
+  );
+}
+
+async function mapBindings(
+  instance: ServiceInstance,
+  repo: Repository,
+): Promise<string[]> {
+  return await Promise.all(
+    instance.bindings.map(async(binding) => {
+      try {
+        return await processBinding(instance, binding, repo);
+      } catch (e) {
+        const bindingIdentifier = instance.instance.serviceInstanceId + "/" +
+          binding.binding.bindingId;
+
+        console.error(
+          `processing binding ${bindingIdentifier} failed with error: ${e}`,
+        );
+
+        const status: OsbServiceInstanceStatus = {
+          status: "failed",
+          description: "Processing the binding failed!",
+        };
+
+        repo.updateInstanceStatus(
+          instance.instance.serviceInstanceId,
+          status,
+        );
+
+        repo.updateBindingStatus(
+          instance.instance.serviceInstanceId,
+          binding.binding.bindingId,
+          status,
+        );
+
+        return bindingIdentifier + ": failed";
+      }
+    }),
+  );
+}
+
+async function processBinding(
+  instance: ServiceInstance,
+  binding: ServiceBinding,
+  repo: Repository,
+): Promise<string> {
+  const bindingIdentifier = instance.instance.serviceInstanceId + "/" +
+    binding.binding.bindingId;
+
+  console.log("starting to process " + bindingIdentifier);
+
+  try {
+    Deno.statSync(
+      `${repo.path}/terraform/${instance.instance.serviceDefinitionId}`,
+    );
+  } catch (e) {
+    if (e instanceof Deno.errors.NotFound) {
+      console.log(
+        `Skipping ${bindingIdentifier}, because no Terraform folder exists for service definition with id ${instance.instance.serviceInstanceId}`,
+      );
+
+      return bindingIdentifier + ": skipped";
+    }
+  }
+
+  const bindingDir =
+    `${repo.path}/instances/${instance.instance.serviceInstanceId}/bindings/${binding.binding.bindingId}`;
+
+  createTerraformWrapper(instance, binding, bindingDir);
+
+  const tfResult = await executeTerraform(bindingDir);
+
+  updateStatusAfterTfApply(tfResult, repo, instance, binding);
+
+  return tfResult.success
+    ? bindingIdentifier + ": successful"
+    : bindingIdentifier + ": failed";
+}
+
+function createTerraformWrapper(
+  instance: ServiceInstance,
+  binding: ServiceBinding,
+  bindingDir: string,
+) {
+  const terraformWrapper = {
+    variable: {
+      client_secret: {
+        description: "Client Secret of the Azure App Registration.",
+        type: "string",
+        sensitive: true,
+      },
```

Comment:
The naming and description is Azure specific. Leaving it like that and making it apparent that currently only Azure is "supported" in the command description is fine for me.

We could then later introduce nullable variables for other supported platforms. The advantage over switching to a single non-platform-specific term would be two have less confusion around naming parameters. Plus if your module needs to access, say, GCP and Azure, that would be possible if we have two variables here.

Reply:
very valid remark. We actually don't need multiple variables here. I just called it platform_secret now. So it can be used for any platform as some kind of secret is always needed.